### ISSUE TITLE
java.lang.IllegalStateException: Reply already submitted

### ISSUE THREAD

**Issue by curt-weber** [harness/ff-flutter-client-sdk#37]
We are seeing crashes related to the latest version (2.2.0) on Android. At the start of each integration test we initialize the `CfClient` and destroy it at the end, and this is likely what is the source of the crashes. Below is the stacktrace:

```
java.lang.IllegalStateException: Reply already submitted
     FATAL EXCEPTION: main
Process: com.kw.app.qa, PID: 20928
java.lang.IllegalStateException: Reply already submitted
	at io.flutter.embedding.engine.dart.DartMessenger$Reply.reply(DartMessenger.java:431)
	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler$1.success(MethodChannel.java:272)
	at io.harness.flutter_cfsdk.FfFlutterClientSdkPlugin$invokeInitialize$1$1.invoke$lambda$0(FfFlutterClientSdkPlugin.kt:107)
	at io.harness.flutter_cfsdk.FfFlutterClientSdkPlugin$invokeInitialize$1$1.$r8$lambda$ZHTo7JY3WpAjOQTDvnO9V2bOCsk(Unknown Source:0)
	at io.harness.flutter_cfsdk.FfFlutterClientSdkPlugin$invokeInitialize$1$1$$ExternalSyntheticLambda0.run(Unknown Source:4)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7918)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
```

**Comment by curt-weber** [comment]
For additional context - this error doesn't surface when we pin the version back to `2.1.3`.

**Comment by erdirowlands** [comment]
Hi @curt-weber thanks for raising this.

I'll investigate this tomorrow and try to reproduce it.  
Can you confirm the Android version your integration test suite is using?

**Comment by curt-weber** [comment]
This occurred on Pixel 7, API Level 33.

**Comment by erdirowlands** [comment]
Thanks @curt-weber that's useful info. Would you mind also raising a support ticket so this can be tracked internally. You can do so by emailing support@harness.io 

I'll come back as soon as I have an update.

**Comment by curt-weber** [comment]
I did already - remembered from last time ;)

**Comment by erdirowlands** [comment]
Hi @curt-weber I have just been able to reproduce this issue - as you say, initializing and destroying the client in quick succession yields 

<img width="1358" alt="Screenshot 2024-06-07 at 17 47 51" src="https://github.com/harness/ff-flutter-client-sdk/assets/23323369/c814afbf-52b5-44b2-abd1-3a3357e5c446">

I've just merged a PR which upgrades the underlying [FF Android SDK](https://github.com/harness/ff-android-client-sdk) to the latest version, hoping that would sort things, as it patches an issue with initialzation.  But soon after merging I was able to reproduce the issue which is still present even after bumping the FF Android SDK.  

Can you confirm:

- is this blocking your production releases?  
- are you seeing this error outside of testing? 


I'll pick this up again on Monday as a priority, as it's nearing the end of the day here in the UK.  But now that I've been able to reproduce it hopefully we can get a resolution.

**Comment by curt-weber** [comment]
It is not blocking production release, we are pinning the old version, but it just leaves debt for us since we can't upgrade this sdk until this is fixed. So don't rush, it seems like you have a handle on it. Appreciate you looking into it, and yes, it is likely related to init/destroy in quick succession which is unlikely to be hit by end users, but integration tests.

**Comment by erdirowlands** [comment]
Thanks for confirming @curt-weber .  I'll work this as a priority and get a fix released as soon as possible.

**Comment by erdirowlands** [comment]
I've got a fix for this in review, and all being well, should be released in a new version tomorrow.  Will keep you posted @curt-weber

**Comment by erdirowlands** [comment]
Hi @curt-weber I've released version 2.2.1 which fixes the issue.  Please give it a try and let me know if it fixes the crash you are seeing.

**Comment by curt-weber** [comment]
We have verified this has been fixed - thanks @erdirowlands
